src/ContributionsFromRenee/WorldScene.js

```diff
@@ -0,0 +1,510 @@
+        frames: [4, 10, 4, 16]
+      }),
+      frameRate: 10,
+      repeat: -1
+    })
+
+    this.anims.create({
+      key: 'right',
+      frames: this.anims.generateFrameNumbers('player', {
+        frames: [4, 10, 4, 16]
+      }),
+      frameRate: 10,
+      repeat: -1
+    })
+
+    this.anims.create({
+      key: 'up',
+      frames: this.anims.generateFrameNumbers('player', {
+        frames: [5, 11, 5, 17]
+      }),
+      frameRate: 10,
+      repeat: -1
+    })
+
+    this.anims.create({
+      key: 'down',
+      frames: this.anims.generateFrameNumbers('player', {
+        frames: [3, 9, 3, 15]
+      }),
+      frameRate: 10,
+      repeat: -1
+    })
+
+    this.anims.create({
+      key: 'idle',
+      frames: [{ key: 'player', frame: 3 }],
+      frameRate: 10
+    })
+  }
+
+  randomizePlayerSpawn(x, y) {
+    // Checks the 9 square area if it's clear to spawn in.
+    let collisionCheck = [
+      map.getTileAt(x - 1, y - 1, true, 'Object Layer').index,
+      map.getTileAt(x, y - 1, true, 'Object Layer').index,
+      map.getTileAt(x + 1, y - 1, true, 'Object Layer').index,
+      map.getTileAt(x - 1, y, true, 'Object Layer').index,
+      map.getTileAt(x, y, true, 'Object Layer').index,
+      map.getTileAt(x + 1, y, true, 'Object Layer').index,
+      map.getTileAt(x - 1, y + 1, true, 'Object Layer').index,
+      map.getTileAt(x, y + 1, true, 'Object Layer').index,
+      map.getTileAt(x + 1, y + 1, true, 'Object Layer').index
+    ]
+
+    if (collisionCheck.every(e => e === -1)) {
+      this.player = new Player(this, x * 16, y * 16, 'player')
+      this.player.body.width = 11
+      this.player.body.height = 11
+      this.player.body.setOffset(2.5, 1)
+    } else {
+      this.randomizePlayerSpawn(x + 1, y + 1)
+    }
+  }
+
+  update(time, delta) {
+    this.player.update(this.cursors)
+  }
+
+  randomizeWorld() {
+    // Fill the floor with random ground tiles
+    groundLayer.weightedRandomize(1, 1, map.width - 2, map.height - 2, [
+      { index: 22, weight: 10 }, // Regular grass
+      { index: 45, weight: 1 }, // One leaf
+      { index: 46, weight: 1 } // Two leaves
+    ])
+
+    // Fill the floor with random, weighted tiles
+    objectLayer.weightedRandomize(1, 1, map.width - 2, map.height - 2, [
+      { index: -1, weight: 50 }, // Empty tile
+      { index: 91, weight: 3 }, // Big Tree
+      { index: 112, weight: 2 } // Small Tree
+    ])
+
+    const x = this.levelConfig.puzzleOptions.x / 16 - 1 //Unit: tiles
+    const y = this.levelConfig.puzzleOptions.y / 16 - 1
+    const width = this.levelConfig.puzzleOptions.width + 2 //Unit: Tiles
+    const height = this.levelConfig.puzzleOptions.height + 2
+
+    // Clear out a rectangle of empty space for sokoban puzzle (top left)
+    objectLayer.fill(-1, x, y, width, height) //clear out any objects for collisions
+    groundLayer.fill(22, x, y, width, height) //yellow tile for puzzlefor plain green: use 22 instead of 85
+  }
+}
```

Comment:
woof. 500 line file is no bueno. How can you break this up? Aim for 100, 200 max.